### PR TITLE
52 link policies tests

### PR DESCRIPTION
**Pull request by ekate**
In this PR I  utilize pytes, pytest-flask and  [pytest-invenio ](https://pytest-invenio.readthedocs.io/en/latest/)  to write tests for template modifications.
-  Added fixture for instance path so we can test custom templates that overwrite InvenioRDM templates
-  Wrote sample test to check footer changes @andrewbattista [did ](https://github.com/nyudlts/ultraviolet/compare/52-link-policies)

### CODE REVIEW

**Review by andrewbattista — APPROVED** [review]
This looks good to me. I don't have a ton of knowledge about good strategies for designing tests that make sense in the context of what is being added. Perhaps we can go over this at a standup, but this seems like a good paradigm to follow

File: tests/test_variables.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    monkeypatch.setenv("SITE_UI_URL", "ultraviolet.dlib.nyu.edu")
+    monkeypatch.setenv("SITE_API_URL", "ultraviolet.dlib.nyu.edu/api")
+    app = create_app()
+    assert app.config.get("APP_ALLOWED_HOSTS") == "ultraviolet.dlib.nyu.edu"
+    assert (
+        app.config.get("SQLALCHEMY_DATABASE_URI")
+        == "postgresql+psycopg2://test:test@somehost.com/test"
+    )
+    assert app.config.get("SITE_UI_URL") == "ultraviolet.dlib.nyu.edu"
+    assert app.config.get("SITE_API_URL") == "ultraviolet.dlib.nyu.edu/api"
```
Comment (Sinclert) [inline tests/test_variables.py:31]: In this test, it seems that multiple values are repeated (once to set them up as env. variables, another time to perform the assertion). I think it would be a good idea to save them in a variable. For instance:

```python
# Some variables assignments here
allowed_host = "ultraviolet.dlib.nyu.edu"
# Some monkey-patching here
monkeypatch.setenv("APP_ALLOWED_HOSTS", allowed_host)
# Some assertions here
assert app.config.get("APP_ALLOWED_HOSTS") == allowed_host
```

File: tests/test_variables.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# Check that default values are assign to configuration variables when not passed through os
+def test_var_noassigned():
+    """Mocking using default configuration"""
+    app = create_app()
+    assert app.config.get("APP_ALLOWED_HOSTS") == ["0.0.0.0", "localhost", "127.0.0.1"]
+    assert (
+        app.config.get("SQLALCHEMY_DATABASE_URI")
+        == "postgresql+psycopg2://nyu-data-repository:nyu-data-repository@localhost/nyu-data-repository"
+    )
+    assert app.config.get("RDM_RECORDS_DOI_DATACITE_ENABLED") == False
```
Comment (Sinclert) [inline tests/test_variables.py:43]: Minor:

In Python, when comparing against **language constants** (`None`, `True`, `False`...), it is better to use `is`, instead of `==`. The reason for this is that `is` compares memory addresses, while `==` compares values. Comparing memory addresses is quicker.

File: tests/ui/conftest.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        app_class=factory.app_class(),
+    )
+
+    return create_ultraviolet_app_ui
+
+
+@pytest.fixture(scope="module")
+def create_app(create_ultraviolet_app_ui):
+    """Flask app fixture."""
+    return create_ultraviolet_app_ui
```
Comment (Sinclert) [inline tests/ui/conftest.py:42]: Why is this needed? This function seem to be just a proxy for another function... 🤔

**Review by Sinclert — CHANGES_REQUESTED** [review]
I left some _minor_ comments to address (optional). In addition:

- We should probably move `pytest`, `flask-pytest` and `pytest-invenio` to the `dev-packages` section of the [Pipfile](https://github.com/nyudlts/ultraviolet/blob/76d0bd215ea449033f7174bff94d54ca805c4027/Pipfile).
- We should probably define a standardized way to run the tests. By default, running `pytest` does not work. This would involve either some comments in the documentation, and / or some `Makefile` rule.

**Review by mnyrop — CHANGES_REQUESTED** [review]
need "running tests" info in docs as part of the PR. then i'll add test runs to the github action. otherwise looks good! (maybe a new page under [/development](https://nyudlts.github.io/ultraviolet/development/))

EDIT: retracting edit in current PR; made a new issue for it #85 so it can happen after #84 is merged.

**Review by mnyrop — APPROVED** [review]

**Review by mnyrop — APPROVED** [review]

File: tests/e2e/conftest.py [hunk 4]
```diff
@@ -20,35 +20,36 @@
...
     )
 
     return create_app_e2e
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope="module")
 def create_app(create_app_e2e):
     """Flask app fixture."""
     return create_app_e2e
```
[the remark this answers is not in the document]
Comment (ekate) [reply tests/e2e/conftest.py:39]: Submit changes

File: tests/test_variables.py (hunk 1 above)
Comment (ekate) [reply tests/test_variables.py]: Submitted changes

File: tests/test_variables.py (hunk 2 above)
Comment (ekate) [reply tests/test_variables.py]: Added changes

File: tests/ui/conftest.py (hunk 3 above)
Comment (ekate) [reply tests/ui/conftest.py:36]: Added changes

**Comment by ekate** [comment]
I  implemented the changes which we re recommended and added `run_test.sh `script. Please take a look and if approved  I will resolve conflict in Pipefile.lock by rebuilding  from new Pipfile and merge

**Comment by Sinclert** [comment]
Thanks for the modifications.

I see there are conflicts with the [Pipfile.lock](https://github.com/nyudlts/ultraviolet/blob/main/Pipfile.lock) file... I think you should have rebased `main` before modifying it on commit https://github.com/nyudlts/ultraviolet/pull/82/commits/505d237cf538cccb2bf41035f813f2cdc7f4a022 . At this point, I would try to override whatever content was on the [Pipfile.lock](https://github.com/nyudlts/ultraviolet/blob/main/Pipfile.lock) filek before with the changes coming from this PR, given that some dependencies have been moved to the `dev-packages` section.

As a last comment: I noticed `selenium` was added both as a `dev-package` and a `package`. Shouldn't it be just for development purposes? 🤔

**Review by Sinclert — APPROVED** [review]
